pkg/config/application_test.go

```diff
@@ -69,21 +55,67 @@ func TestNewApplicationGroupFailure(t *testing.T) {
 	tests := []struct {
 		testName       string
 		applicationMod func(*Application)
-		expectedErr    error
+		expectedErr    string
 	}{
 		{
 			testName: "When application group policy is empty",
 			applicationMod: func(a *Application) {
 				a.Policies = nil
 			},
-			expectedErr: errors.New("no policies defined"),
+			expectedErr: "no policies defined",
 		},
 		{
-			testName: "When adding policies to application group",
-			applicationMod: func(a *Application) {
-				a.Organizations[0].Policies = nil
+			testName: "When no Admins policies are defined",
+			applicationMod: func(application *Application) {
+				delete(application.Policies, AdminsPolicyKey)
+			},
+			expectedErr: "no Admins policy defined",
+		},
+		{
+			testName: "When no Readers policies are defined",
+			applicationMod: func(application *Application) {
+				delete(application.Policies, ReadersPolicyKey)
+			},
+			expectedErr: "no Readers policy defined",
+		},
+		{
+			testName: "When no Writers policies are defined",
+			applicationMod: func(application *Application) {
+				delete(application.Policies, WritersPolicyKey)
+			},
+			expectedErr: "no Writers policy defined",
+		},
+		{
+			testName: "When ImplicitMetaPolicy rules' subpolicy is missing",
+			applicationMod: func(application *Application) {
+				application.Policies[ReadersPolicyKey].Rule = "ALL"
+			},
+			expectedErr: "invalid implicit meta policy rule: 'ALL': expected two space separated " +
```

Comment:
nit: there's no need to separate these long lines to satisfy the "long line linter." While it doesn't matter too much in test, it does make it more difficult to find messages in production code and it's easier to drop spaces between words.